source/javascripts/controllers/_EnvVarsController.js.erb

```diff
@@ -186,6 +197,42 @@ angular.module("BitriseWorkflowEditor").controller("EnvVarsController", function
 		return errors.length > 0 ? stringService.errorMessageFromErrors(errors) : null;
 	};
 
+	viewModel.valueVisibilityGetterSetterForEnvVar = function(secret) {
+		return function(shouldShowValue) {
+			if (shouldShowValue !== undefined) {
+				if (shouldShowValue && !secret.customProperties.progress.isInProgress) {
+					getSecretValue(secret);
+				}
+
+				secret.shouldShowValue = shouldShowValue;
+			}
+
+			return secret.shouldShowValue;
+		};
+	};
+
+	function getSecretValue(secret) {
+		secret.customProperties.progress.start("<%= data[:strings][:env_vars][:get_secret_env_value_progress][:in_progress] %>");
+
+		appService.getSecretValue(secret).then(function() {
+			secret.customProperties.progress.success();
+		}, function(error) {
+			secret.customProperties.progress.error(error);
+		});
+	}
+
+	viewModel.valueGetterSetterForEnvVar = function(envVar) {
+		return function(value) {
+			return (envVar.value() !== null && envVar.shouldShowValue) ? envVar.value(value) : "******";
+		};
+	};
+
+	viewModel.makeEnvVarProtected = function(envVar) {
+		Popup.showNoticePopup("<%= data[:strings][:env_vars][:alert_make_it_protected_popup][:title] %>", "<%= data[:strings][:env_vars][:alert_make_it_protected_popup][:details] %>", "<%= data[:strings][:env_vars][:alert_make_it_protected_popup][:ok_button_title] %>").then(
+			envVar.shouldShowValue = false,
+			envVar.isProtected(!envVar.isProtected())
```

Comment:
Set it to `true`, not the opposite of the current state. It will be the same, but `true` tells more accurately what we're trying to do here.